### PR TITLE
Bugfix FXIOS-11815 ⁃ [New Toolbar experiment] [iPad] - Tab name is not correctly displayed when set to 'Empty New Tab' or 'Homepage'

### PR DESCRIPTION
**Pull request by dicarobinho**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-11815)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/25769)

## :bulb: Description
Use New Tab title, when homepage is blank

## :movie_camera: Demos
IMG_0083
IMG_0082

| Before | After |
| - | - |
|  |  |
|  |  |
|  |  |

<details>
<summary>Demo</summary>

</details>

## :pencil: Checklist
- [x] I filled in the ticket numbers and a description of my work
- [x] I updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] I ensured unit tests pass and wrote tests for new code
- [x] If working on UI, I checked and implemented accessibility (Dynamic Text and VoiceOver)
- [ ] If adding telemetry, I read the [data stewardship requirements](https://github.com/mozilla-mobile/firefox-ios/wiki/Adding-Glean-Telemetry-Events) and will request a data review
- [ ] If adding or modifying strings, I read the [guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/How-to-add-and-modify-Strings) and will request a string review from l10n
- [ ] If needed, I updated documentation and added comments to complex code

### CODE REVIEW

File: firefox-ios/Client/TabManagement/Tab.swift
Comment (FilippoZazzeroni) [inline firefox-ios/Client/TabManagement/Tab.swift:258]: Can we add tests for this method i see `TabTests` is missing it ?

File: firefox-ios/Shared/Strings.swift [hunk 1]
```diff
@@ -3799,6 +3799,11 @@ extension String {
             tableName: nil,
             value: "Private",
             comment: "The title on the button to look at private tabs.")
+        public static let TabsSelectorBlankTabsTitle = MZLocalizedString(
```
Comment (thatswinnie) [inline firefox-ios/Shared/Strings.swift:3838]: As this string is not translated into all languages right away we should put all the changes that use the new string behind a feature flag.

Comment (FilippoZazzeroni) [reply firefox-ios/Shared/Strings.swift:3838]: Thanks Winnie for looking into this i didn't spot it

**Comment by dicarobinho** [comment]
Since this PR has a string dependency until it is translated, will not merge it. I will create another [PR](https://github.com/mozilla-mobile/firefox-ios/pull/32167) with only the necessary string and after translation is done, will merge this PR.
cc: @thatswinnie

File: firefox-ios/Shared/Strings.swift [hunk 2]
```diff
@@ -3799,6 +3799,11 @@ extension String {
...
+            key: "TabTray.TabsSelectorBlankTabsTitle.v149",
```
Comment (lmarceau) [inline firefox-ios/Shared/Strings.swift:3839]: I know this will be opened in another PR, but note that this needs to ride the 150 train, not 149

File: firefox-ios/Client/Frontend/Browser/Tabs/Views/TabCell.swift [hunk 3]
```diff
@@ -117,17 +117,25 @@ final class TabCell: UICollectionViewCell,
 
         animator?.animateBackToCenter()
 
-        titleText.text = tabModel.tabTitle
-        accessibilityLabel = getA11yTitleLabel(tabModel: tabModel)
+        // When homepage is blank, we need to use firefoxFavIcon and
+        // "New Tab" title
+        if tabModel.url == nil {
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Browser/Tabs/Views/TabCell.swift]: I think this logic could be handled in the VC rather than the `UITableViewCell`, meaning the `TabModel` would provide the required title directly

File: firefox-ios/Client/Frontend/Browser/Tabs/Views/TabCell.swift
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Browser/Tabs/Views/TabCell.swift]: This change should be done for the `ExperimentTabCell` as well I think. `TabCell` will be removed soon when we align the iPad and iPhone tab tray experience cc: @thatswinnie

File: firefox-ios/Shared/Strings.swift (hunk 2 above)
Comment (dicarobinho) [reply firefox-ios/Shared/Strings.swift:3839]: Strings PR has been merged last week so v149 should be fine.

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/TabCellTests.swift [hunk 4]
```diff
@@ -32,22 +32,22 @@ class TabCellTests: XCTestCase {
     func testConfigureTabAXLabel() {
```
Comment (FilippoZazzeroni) [inline firefox-ios/firefox-ios-tests/Tests/ClientTests/TabCellTests.swift:32]: nit: could we add a test for the `configure` when the `newTabTitle` is not nil

**Review by FilippoZazzeroni — APPROVED** [review]
The changes looks good to me @dicarobinho the only things we should wait for `149` translations import before merging this.

**Comment by dicarobinho** [comment]
Thank you @thatswinnie! For sure there is required a consistency between iPad and iPhone. I will fix that.

**Comment by megandaniel** [comment]
> With this change there is now incosistencies between iPhone and iPad. Please check with product and design that this is what we want. 
> Also the favicon we are using seems pretty old (it's pngs in different scales instead of a pdf) - do we maybe want to update it? @topotropic

Hi @thatswinnie ! Nicole passed this along to me – I think it makes sense to update mobile as well (string + favicon) for consistency. I have that mocked up here FYI: (https://www.figma.com/design/BVqEYexibOA2betNVUMrfl/Mobile-Assembly-File-2026?node-id=9970-129214&t=50TCOo7R0KGyipBj-1). It's been shared with Andy, pending his feedback.

**Comment by dicarobinho** [comment]
IMG_9042

**Comment by thatswinnie** [comment]
DO NOT MERGE UNTIL 149.0.

This PR depends on https://github.com/mozilla-mobile/firefox-ios/pull/32167 merged in 148.1.

**Comment by afurlan-firefox** [comment]
Looks good! Both updates (iPhone and iPad).
